parser.go

```diff
@@ -1587,6 +1587,7 @@ func (p *parser) parseAstField(pkgPath, pkgName string, structSchema *SchemaObje
 					}
 					structSchema.Properties.Set(propertyName, refPropertySchema)
 				}
+				structSchema.Required = append(structSchema.Required, refSchema.Required...)
```

Comment:
This is the only functional change. Everything else is test/fixture updates :P

Reply:
And there's no way `refSchema` could be empty? Or it doesn't matter anyway if it is as it just won't have and `required` properties to unpack?

Reply:
Yeah, we're still within the `if ok {` block, which I believe is guarding against `refSchema` not having been found in `p.KnownIDSchema`.

Reply:
I guess it's possible that `nil` was inserted into `KnownIDSchema` for the given id, and we'll get a panic if that's the case - we probably want that though because that would be a bug.